<?xml version="1.0"?>
<launch>

<arg name="load_gripper" default="false" />
  <include file="$(find franka_control)/launch/franka_control.launch" >
    <arg name="robot_ip" value="172.16.0.12" />
    <arg name="load_gripper" value="$(arg load_gripper)" />
  </include>

  <rosparam command="load" file="$(find franka_example_controllers)/config/franka_example_controllers.yaml" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen"  args="kth_joint_pose_effort_interface_controller"/>

    <node pkg="tf"
        type="static_transform_publisher"
        name="robot_world_tf"
        args="0 0 0 0 0 0  world panda_link0 10" />

  <arg name="stop_at_shutdown" default="true" />
  <arg name="arm_id"      default="panda" />
  <arg name="joint_names" default="[$(arg arm_id)_finger_joint1, $(arg arm_id)_finger_joint2]" />
  <node name="fg_franka_gripper" pkg="franka_gripper" type="fg_franka_gripper_node" output="screen">
    <param name="robot_ip" value="172.16.0.12"/>
    <param name="stop_at_shutdown" value="$(arg stop_at_shutdown)" />
    <rosparam param="joint_names" subst_value="true">$(arg joint_names)</rosparam>
  </node>
</launch>